module/admin/helper.js

```diff
@@ -128,4 +140,455 @@ module.exports = class AdminHelper {
+				let resourceIdsWithType = []
+				let surveyIds = []
+
+				// Handle deletion if the resource type is PROGRAM
+				if (resourceType === CONSTANTS.common.PROGRAM) {
+					const programFilter = {
+						_id: resourceId,
+						tenantId: tenantId,
+						isAPrivateProgram: false,
+					}
+
+					// Fetch the program to ensure it exists
+					const programDetails = await programsQueries.programsDocument(programFilter, ['components'])
+					if (!programDetails || !programDetails.length) {
+						throw {
+							status: HTTP_STATUS_CODE.bad_request.status,
+							message: CONSTANTS.apiResponses.PROGRAM_NOT_FOUND,
+						}
+					}
+
+					// Initialize projectTemplate and solution ID holders
+					let projectTemplateIds
+					let solutionIds
+					if (programDetails[0].components) {
+						// Extract solution IDs from program components
+						solutionIds = programDetails[0].components.map((component) => component._id)
+
+						const solutionFilter = {
+							_id: { $in: solutionIds },
+							tenantId,
+						}
+						// Get solution documents to retrieve associated project templates
+						const solutionDetails = await solutionsQueries.solutionsDocument(solutionFilter, [
+							'projectTemplateId',
+						])
+
+						const foundSolutionIds = solutionDetails.map((doc) => String(doc._id))
+
+						const missingSolutionIds = solutionIds.filter((id) => !foundSolutionIds.includes(String(id)))
+
+						// Push them into surveyIds
+						surveyIds.push(...missingSolutionIds)
+
+						if (surveyIds.length > 0) {
+							const deleteResponse = await surveyService.deleteSolutionResource(
+								surveyIds,
+								CONSTANTS.common.SOLUTION,
+								tenantId,
+								orgId,
+								deletedBy
+							)
+
+							surveyCount += deleteResponse.data.result.surveyCount || 0
+							surveySubmissionCount += deleteResponse.data.result.surveySubmissionCount || 0
+							observationCount += deleteResponse.data.result.observationCount || 0
+							observationSubmissionCount += deleteResponse.data.result.observationSubmissionCount || 0
+						}
+
+						await solutionsQueries.deleteSolutions(solutionFilter)
+						solutionDeletedCount++
+						if (solutionIds && solutionIds.length) {
+							for (const Id of solutionIds) {
+								resourceIdsWithType.push({ id: Id, type: CONSTANTS.common.SOLUTION })
+							}
+						}
+						// Extract project template IDs from solutions
+						projectTemplateIds = solutionDetails.map((projectTemplateIds) => {
+							return projectTemplateIds.projectTemplateId
+						})
+					}
+					// Add main program ID to deletion list
+					resourceIdsWithType.push({ id: resourceId, type: CONSTANTS.common.PROGRAM })
+
+					// Delete all projects linked to the solutions
+					let projecFilter = {
+						solutionId: { $in: solutionIds },
+						isAPrivateProgram: false,
+					}
+					let deletedProjectIds = await projectQueries.deleteProjects(projecFilter)
+
+					if (deletedProjectIds.deletedCount > 0) {
```

Comment:
@MallanagoudaB what is the  issue in assigning deletedCount without the if condition?